nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestFlattenJson.java

```diff
@@ -0,0 +1,520 @@
+import org.apache.nifi.util.MockFlowFile;
+import org.apache.nifi.util.TestRunner;
+import org.apache.nifi.util.TestRunners;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class TestFlattenJson {
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    @Test
+    void testFlatten() throws JsonProcessingException {
+        final TestRunner testRunner = TestRunners.newTestRunner(FlattenJson.class);
+        final String json = "{\n" +
+                "    \"test\": {\n" +
+                "        \"msg\": \"Hello, world\"\n" +
+                "    },\n" +
+                "    \"first\": {\n" +
+                "        \"second\": {\n" +
+                "            \"third\": [\n" +
+                "                \"one\",\n" +
+                "                \"two\",\n" +
+                "                \"three\",\n" +
+                "                \"four\",\n" +
+                "                \"five\"\n" +
+                "            ]\n" +
+                "        }\n" +
+                "    }\n" +
+                "}";
+        final Map parsed = (Map) baseTest(testRunner, json, 2);
+        assertEquals(parsed.get("test.msg"), "Hello, world", "test.msg should exist, but doesn't");
+        assertEquals(parsed.get("first.second.third"),
+                Arrays.asList("one", "two", "three", "four", "five"),
+                "Three level block doesn't exist.");
+    }
+
+    private Object baseTest(TestRunner testRunner, String json, int keyCount) throws JsonProcessingException {
+        return baseTest(testRunner, json, Collections.emptyMap(), keyCount);
+    }
+
+    private Object baseTest(TestRunner testRunner, String json, Map attrs, int keyCount) throws JsonProcessingException {
+        testRunner.enqueue(json, attrs);
+        testRunner.run(1, true);
+        testRunner.assertTransferCount(FlattenJson.REL_FAILURE, 0);
+        testRunner.assertTransferCount(FlattenJson.REL_SUCCESS, 1);
+
+        final List<MockFlowFile> flowFiles = testRunner.getFlowFilesForRelationship(FlattenJson.REL_SUCCESS);
+        final byte[] content = testRunner.getContentAsByteArray(flowFiles.get(0));
+        final String asJson = new String(content);
+        if (asJson.startsWith("[")) {
+            final List parsed;
+            parsed = mapper.readValue(asJson, List.class);
+            assertEquals(keyCount, parsed.size(), "Too many keys");
+            return parsed;
+        } else {
+            final Map parsed;
+            parsed = mapper.readValue(asJson, Map.class);
+            assertEquals(keyCount, parsed.size(), "Too many keys");
+            return parsed;
+        }
+    }
```

Comment:
All private methods should be moved to the end of the class, following standard conventions.

Reply:
Moved it!